### PR TITLE
Request context

### PR DESCRIPTION
**Pull request by wallin**
Facilitate sending requests to Castle from an async environment by introduce a RequestContext object

### CODE REVIEW

**Comment by lluft** [comment]
@wallin fixed the last little bug. Manually tested and working all smooth. Ready for final review and release

File: README.md [hunk 1]
```diff
@@ -28,6 +28,47 @@ Configure the library with your Castle API secret.
...
+// Pass the context to track, identify or authenticate
+Castle::track(array(
+  'name' => $event['name'],
+  'user_id' => $event['user_id'],
+  'context' => json_decode($context)
+));
+```
+
+
+// Send the request to Castle
```
Comment (wallin) [inline README.md]: Remove this?

File: test/TestTransport.php [hunk 2]
```diff
@@ -12,18 +12,19 @@ class Castle_RequestTransport
 
   private static $lastRequest = array();
 
-  public function send($method, $url, $body=null, $headers=array()) {
+  public function send($method, $url, $payload) {
     if (empty(self::$params)) {
       self::setResponse(200, '{}');
     }
+    $headers = array('Content-Type' => 'application/json');
```
Comment (wallin) [inline test/TestTransport.php]: How about sending content length? If I remember correctly, it was needed in som cases

Comment (lluft) [reply test/TestTransport.php]: Better safe than sorry. Will add :)

**Pull request merged**